src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIntegTestBase.java

```diff
@@ -223,62 +224,16 @@ public void ensurePaAndRcaEnabled() throws Exception {
                 PerformanceAnalyzerClusterSettingHandler.checkBit(state, PerformanceAnalyzerFeatureBits.RCA_BIT.ordinal()));
     }
 
-    @Test
-    public void checkMetrics() throws Exception {
-        ensurePaAndRcaEnabled();
-        final String[] jsonString = new String[1];
-        WaitFor.waitFor(() -> {
-            Request request = new Request("GET",
-                    "/_opendistro/_performanceanalyzer/metrics/?metrics=Disk_Utilization&agg=max&dim=&nodes=all");
-            Response resp = paClient.performRequest(request);
-            Assert.assertEquals(HttpStatus.SC_OK, resp.getStatusLine().getStatusCode());
-            jsonString[0] = EntityUtils.toString(resp.getEntity());
-            JsonNode root = mapper.readTree(jsonString[0]);
-            for (Iterator<JsonNode> it = root.elements(); it.hasNext(); ) {
-                JsonNode entry = it.next();
-                JsonNode data = entry.get(TestUtils.DATA);
-                if (data.get(TestUtils.FIELDS) == null) {
-                    return false;
-                }
-            }
-            return jsonString[0] != null && !jsonString[0].isEmpty();
-        }, 1, TimeUnit.MINUTES);
-        logger.info("jsonString is {}", jsonString[0]);
-        JsonNode root = mapper.readTree(jsonString[0]);
-        root.forEach( entry -> {
-            JsonNode data = entry.get(TestUtils.DATA);
-            Assert.assertEquals(1, data.get(TestUtils.FIELDS).size());
-            JsonNode field = data.get(TestUtils.FIELDS).get(0);
-            Assert.assertEquals(TestUtils.M_DISKUTIL, field.get(TestUtils.FIELD_NAME).asText());
-            Assert.assertEquals(TestUtils.DOUBLE_TYPE, field.get(TestUtils.FIELD_TYPE).asText());
-            JsonNode records = data.get(TestUtils.RECORDS);
-            Assert.assertEquals(1, records.size());
-            records.get(0).forEach(record -> Assert.assertTrue(record.asDouble() >= 0));
-        });
-    }
-
-    @Test
-    public void testRcaIsRunning() throws Exception {
-        ensurePaAndRcaEnabled();
-        WaitFor.waitFor(() -> {
-            Request request = new Request("GET", "/_opendistro/_performanceanalyzer/rca");
-            try {
-                Response resp = paClient.performRequest(request);
-                return Objects.equals(HttpStatus.SC_OK, resp.getStatusLine().getStatusCode());
-            } catch (Exception e) { // 404, RCA context hasn't been set up yet
-                return false;
-            }
-        }, 2, TimeUnit.MINUTES);
-    }
 
     @After
     public void closePaClient() throws Exception {
         ESRestTestCase.closeClients();
         paClient.close();
         LOG.debug("AfterClass has run");
+        cleanUpCluster();
```

Comment:
This is dangerous, I explicitly overrode preserveClusterUponCompletion() so that this would destroy existing clusters that we point the test to. Maybe we could have a sysprop flag here and have  preserveClusterUponCompletion() return the value of the flag if this is needed?

Reply:
Did you mean "wouldn't"? Because otherwise it doesn't add up as we're overriding to preserve the cluster settings in the base class.

That being said, this call to cleanUpCluster() was a no-op all along. I've removed it now. Thanks for pointing it out!